src/api/firebase.js

```diff
@@ -85,11 +86,43 @@ export async function updateItem(listId, key, checkedState) {
 		const newDateLastPurchased = checkedState
 			? new Date()
 			: doc.data().dateLastPurchased;
+
 		updateDoc(docRef, {
 			totalPurchases: newTotalPurchases,
 			dateLastPurchased: newDateLastPurchased,
 			checked: checkedState,
 		});
+
+		const currentDate = new Date();
+		const updatedDateNextPurchased = new Date(
+			doc.data().dateNextPurchased.toDate(),
+		);
+		const updatedDateCreated = new Date(doc.data().dateCreated.toDate());
+		const formattedDateLastPurchased = new Date(newDateLastPurchased);
+
+		// gets last estimated interval
+		const previousEstimate =
+			updatedDateNextPurchased === null
+				? getDaysBetweenDates(updatedDateNextPurchased, updatedDateCreated)
+				: getDaysBetweenDates(
+						updatedDateNextPurchased,
+						formattedDateLastPurchased,
+				  );
+
+		const daysSinceLastPurchase =
+			updatedDateNextPurchased === null
+				? getDaysBetweenDates(currentDate, updatedDateCreated)
+				: getDaysBetweenDates(currentDate, formattedDateLastPurchased);
+
+		const daysTilNextPurchase = calculateEstimate(
+			previousEstimate,
+			daysSinceLastPurchase,
+			newTotalPurchases,
+		);
+
+		updateDoc(docRef, {
+			dateNextPurchased: getFutureDate(daysTilNextPurchase),
+		});
```

Comment:
While this code isn't giving any errors, it wasn't estimating the dateNextPurchased properly. 
I believe we need to create a new key for the javascript object that will store the previousEstimate/daysTillNextPurchase. 

And then also, we can do the updateDoc() function all at once in the end (instead of calling it twice).

Here's my suggestion: 

```suggestion
		const previousEstimate = doc.data().daysTillNextPurchase; //this gets the previousEstimate from the datebase. if it doesn't already exist, this returns undefined (which will be interpreted as 14 by the calculateEstimate() function)
		const today = new Date();
		const dateCreated = new Date(doc.data().dateCreated.toDate()); //converts dateCreated to from timestamp to JS Date
		const dateLastPurchased = doc.data().dateLastPurchased === null 
			? dateCreated
			: new Date(doc.data().dateLastPurchased.toDate()); 
		//gets the date last purchased (converted from timestamp to JS date OR dateCreated if it doesn't exist
		
		const daysSinceLastPurchase = getDaysBetweenDates(today, dateLastPurchased);

		const daysTillNextPurchase = calculateEstimate(
			previousEstimate, //I believe this value MUST come from the database/memory and cannot be calculated here
			daysSinceLastPurchase,
			newTotalPurchases,
		);

		updateDoc(docRef, {
			totalPurchases: newTotalPurchases,
			dateLastPurchased: newDateLastPurchased,
			checked: checkedState,
			dateNextPurchased: getFutureDate(daysTillNextPurchase),
			daysTillNextPurchase: daysTillNextPurchase, // this will store the calculatedEstimate (and at the next purchase, it will become the previousEstimate
		});
```

Reply:
> previousEstimate, //I believe this value MUST come from the database/memory and cannot be calculated here

@FuentesNina could you say more about this? Was there something you read or noticed that led you to that conclusion?

I haven't tried it out, but I would have thought that you should be able to use the `dateNextPurchased` Date field from the database to calculate `daysTillNextPurchase` in memory (here in the `updateItem` function)

(Also +1 to the suggestion to only call updateDoc once)

Reply:
@meganesu  you're correct. the previous date can be calculated without storing daysTillNextPurchase in the database. I'm going to comment on the latest commit.